services/github/src/utils.js

```diff
@@ -173,6 +173,94 @@ export const tailLog = (log, maxLines = 250, maxPayloadLength = 30000) => {
     .join('\n');
 };
 
+/**
+ *
+ * @param {string} logString
+ * @param {number} maxPayloadLength
+ * @param {number} tailLines
+ * @returns string or null
+ */
+export const extractTailLinesFromLog = (logString, maxPayloadLength, tailLines) => {
+
+  if(tailLines === 0) {return null;}
+
+  const tailLog = logString.split("\n").slice(-tailLines).join("\n");
+
+  if(logString.length <= maxPayloadLength) {
+    return tailLog;
+  }
+
+  let tailLogMaxPayload = logString.slice(-maxPayloadLength);
+  const newLinePosition = tailLogMaxPayload.indexOf('\n');
+  tailLogMaxPayload = tailLogMaxPayload.slice(newLinePosition + 1);
+
+  return tailLog.length <= tailLogMaxPayload.length ? tailLog : tailLogMaxPayload;
+};
+
+/**
+ *
+ * @param {string} logString
+ * @param {number} maxPayloadLength
+ * @returns string
+ */
+export const extractHeadLinesFromLog = (logString, maxPayloadLength) => {
+
+  if(logString.length <= maxPayloadLength) {return logString;}
+
+  const headLog = logString.slice(0, maxPayloadLength);
+  const lastNewLinePosition = headLog.lastIndexOf('\n');
+  return headLog.substring(0, lastNewLinePosition);
+};
+
+/**
+ * Github checks API call is limited to 64kb
+ * @param {string} log
+ * @param {number} headLines
+ * @param {number} tailLines
+ * @param {number} maxPayloadLength
+ * @returns string
+ */
+export const extractLog = (log, headLines = 20, tailLines = 200, maxPayloadLength = 30000) => {
+
+  if(headLines < 0 && tailLines < 0) {return extractLog(log, -1 * tailLines, -1 * headLines, maxPayloadLength);}
+
```

Comment:
handles cases for negative inputs to this function

Reply:
Do you think it will be useful/used anywhere?

Reply:
I don't think anyone would trigger it knowingly, thought it would be good behaviour if anyone would trigger with negative inputs, it should cycle from back. 

Will remove this check